bench.c

```
@@ -0,0 +1,113 @@
+                const char *longer_str = len_a > len_b ? str_a : str_b;
+                identical = (longer_str[comp_len] == '/');
+            }
+        }
+    }
+end:
+    return identical;
+}
+
+static Link *Link_new(const char *linkname, LinkType type) {
+    Link *link = calloc(1, sizeof(Link));
+    strncpy(link->linkname, linkname, NAME_MAX);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)